adapter/internal/discovery/xds/server.go

```diff
@@ -277,16 +277,23 @@ func UpdateAPI(vHost string, apiProject model.ProjectAPI, environments []string)
 		return nil, err
 	}
 
-	if apiYaml.APIType == constants.HTTP || apiYaml.APIType == constants.WEBHOOK {
-		err = mgwSwagger.GetMgwSwagger(apiProject.OpenAPIJsn)
-
-		// Set the following in case they were overridden by the above line
-		mgwSwagger.SetID(apiYaml.ID)
-		mgwSwagger.SetName(apiYaml.Name)
-		mgwSwagger.SetVersion(apiYaml.Version)
-		if err != nil {
-			return nil, err
-		}
+	err = apiProject.APIYaml.ValidateAPIType()
+	if err != nil {
+		return nil, err
+	}
+
+	err = mgwSwagger.GetMgwSwagger(apiProject.APIDefinition)
```

Comment:
hope we have handled the cases where API project does not contains API def. like in websocket apis.

Reply:
From this PR onwards, websocket APIs too will have an API definition file. This would be of type `asyncapi`.